packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -749,6 +750,9 @@ function emitBasicOperation(
   });
 
   for (const param of httpOperation.parameters.parameters) {
+    if (isCustomClientRequestIdParam(param)) {
```

Comment:
In modular we just ignore this custom header parameter.

Reply:
In the original issue, it also mentioned a return-client-request-id header, I don't see any logic is handling it in the response ?

Reply:
Also, have we agreed to completely ignore this custom header in Modular ? what if customers set it via options.headers['client-request-id'] ?  And if customer has specify a diffrent header key than client-request-id, should we ignore that header key instead of this one ? 
Is it possible there are other headers that we want customers to ignore ? and client-request-id might be just one of them, are we going to have isXXXParam for all of them ?

Reply:
I think the general guideline would be - the custom header would take the same role as original `x-ms-client-request-id`. If we ignore the latter we are supposed to ignore for custom one.

And I added test cases to demo on how to set other client header and how to override request id in modular. https://github.com/Azure/autorest.typescript/blob/bac92fb61ad9bd83938754139ada2935f3f881d3/packages/typespec-ts/test/modularIntegration/customClientRequestId.spec.ts